### ISSUE TITLE
Add support to M18x R2 Computer

### ISSUE THREAD

**Issue by Lukw3** [rsm-gh/akbl#86]
/!\ IF YOU WANT YOUR BUG TO BE TREATED FILL THIS TEMPLATE AND DO NOT MODIFY IT. /!\

      Also, before submitting any bug download and
            install the last version of akbl.

# System information

+ Distribution name: Blackarch
+ Distribution version: 2020.06
+ Python3 version: `python3 --version`: 3.8.5
+ akbl version: `akbl -v`: 2019.05.26
+ Computer model: M18x R2
+ Daemon status: `akbl --daemon-is-on`: True

# Bug description

Output of lsusb:

Device: ID 187c:0518 Alienware Corporation 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.10
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x187c Alienware Corporation
  idProduct          0x0518 
  bcdDevice            0.00
  iManufacturer           1 Alienware
  iProduct                2 M18x
  iSerial                 3 1.1.3
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0029
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower               64mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.01
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      56
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              10
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              10
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0000
  (Bus Powered)

I am running the Alienware 17R? config right now. it seems to work pretty good, other then some weird shit with the colors. it would be nice to get the proper config..
Thanks,
  --Lukwe

**Comment by rsm-gh** [comment]
Hello @Lukw3 ,

Thanks for your feedback. I just added your configuration basing me from the M18x. Can you please make the install and check if the zones work?

Concerning the colors it may be a bug of akbl. We will check that one that all the zones work.


regards

**Comment by Lukw3** [comment]
Thanks a lot! 

Speaker: Right, seems to control the left speaker, not a huge problem tho.. Also, Speaker Left seems to control the right speaker as well.

The head on the back of the computer seems to be un-effected (it just glows white)by the color i set it too, even if i turn all of the lights of it continues to glow.

The Alienware logo is a bit finicky, when i set it to yellow, it turns off. and yet when i set it to red it seems to work, but only after a bit of delay.. (like 2 mins)

Keyboard Right, Left, Right-Center and Left-Center all seem to be working, there doesn't seem to be an option for the AlienFx keys, but thats okay.

the media bar does not work, I'm assuming that means the controls for the volume and such at the top-right..

The power button alien-head type thing also does not seem to work.

The Touch pad also does not work, i hopped back to the 17R? Config just for the heck of it and it works fine :D 

Anyways, i hope this helps. Thanks for putting the time in to making this awesome program. Let me know if theres anything i can do to contribute.

Thanks,
  --Lukw3

**Comment by rsm-gh** [comment]
Hello @Lukw3 ,

Thanks for your feedback, I corrected the config with the M17R? so some of the problems should be okay. If it is possible it would be nice to have an exhaustive feedback of the problems that you still having to try to fix them.

Normally this work is done trough the block testing window, you can iterate all the proposed blocks and try to find the missing areas like the media bar that it is not working for you (at the top left, for indicating the caps status, etcc).

I may take some time to answer but don't hesitate to write me back :) 

rsm~

**Comment by Lukw3** [comment]
Hey @rsm-gh !

The speaker lights are now properly aligned.. :)
I have done some messing around with the block-testing window, but i cannot find the media bar after trying  regions 1 - 4194303. 
Same thing above with the head on the back (Maybe its not meant to be controllable at all?)
Also the touch-pad now works. :+1: 

One idea i have is to try to emulate the windows drivers with wine or something.. than somehow capture the traffic to the lights. idk.. :)

I'm not sure what i could to provide 'exhaustive feedback' but I'll certainly do anything to contribute! 

Anyways, Thanks for the help.. Let me know if there's anything i can do :)

Thanks
  --Lukwe
